feat: Max onboarding

kappa90 opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem
Max's onboarding would get stuck when Perplexity didn't have a good answer when asked for the user's company's description. This PR solves that by adding a new node `MemoryOnboardingEnquiryNode` which recursively asks the user for information about their product/features/business model, and then saves the information to the core memory.

## Changes
- `MemoryInitializerInterruptNode` now only interrupts the conversation
- new `MemoryOnboardingEnquiryNode` to enquire about the user's company, with a maximum of 3 questions asked to the user before moving on. The questions can be in addition to Perplexity's answers, if the model thinks that's not enough.
- The onboarding button is now the only available option when talking to Max for the first time

![image](https://github.com/user-attachments/assets/efafac33-816e-47b2-95a4-6f20d6406ce1)
![image](https://github.com/user-attachments/assets/d86ed192-6f14-4b70-b886-2988e8224c8b)


## Does this work well for both Cloud and self-hosted?
Yes

## How did you test this code?
Chatted with Max + added/fixed tests

Commit:
feat(max): new onboarding

Commit:
fix: tests

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix: tests

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

Commit:
Merge remote-tracking branch 'origin/max/onboarding' into max/onboarding

Commit:
fix: test

Twixes committed:
Merge branch 'master' into max/onboarding

Twixes reviewed (changes requested):
_This review is from before our sync, so we have some new ideas now._

This is the right direction, but doesn't quite _spark joy_ yet. The questions and the way Max asks them aren't as relevant as they could be. E.g. here's me as a first-time user who hasn't onboarded before:

<img width="377" alt="bad" src="https://github.com/user-attachments/assets/e8f030de-0ff9-4ddc-9f8d-c30e209784ea" />

There's three issues in this experience:
1. ~~Max's answer doesn't flow from my actual question, it's robotic and generic - a human would say something like "I'm not yet sure how to define top users, as I don't know much about your product. Is it okay if I ask a few questions? [proceeds to ask questions]"~~ Actually there shouldn't be a text box, as we've resolved in the sync.
2. While the first message is too generic, the questions seem too specific actually. E.g. I'll never list out "the specific features" of my product, because that list comprises ALL the features, of which there are lots of them. We should ask maybe about the most important features, key aspects.
3. This onboarding is stubborn as hell, if I say something that doesn't answer the question, it just keeps going with questions instead of responding.

Also +1 to all of Georgiy's points.

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

Commit:
feat(max): onboarding v.0.2

Commit:
Merge remote-tracking branch 'origin/max/onboarding' into max/onboarding

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

Commit:
fix: add variable

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
fix: better prompt, small bugs fixed

Commit:
Merge remote-tracking branch 'origin/max/onboarding' into max/onboarding

Commit:
fix: better check

Twixes committed:
Remove redundant "the"

Twixes committed:
Remove product_description from demo data

We can still keep some vestigial support for `Project.product_description` for earlier users of Max, but we should drop it some time after launch, and definitely shouldn't be including in the demo data anymore. Instead, the demo data should have either a core memory pre-set, or nothing. Opting for nothing here, so that we experience the onboarding actually.

Twixes reviewed (commented):
Works quite well, and there's no _blocker_. Did end up with some comments after going through the code

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

kappa90 commented on ee/hogai/graph/memory/nodes.py:
```diff
@@ -79,31 +82,72 @@ def _retrieve_context(self):
+            return "continue"
+
+        last_message = state.messages[-1]
+        if isinstance(last_message, HumanMessage) and last_message.content == ONBOARDING_INITIAL_MESSAGE:
+            return "memory_onboarding"
+        return "continue"
```
No, I'm pretty sure the trace you shared happens only when you ask for an insight, and we haven't run the onboarding yet. Unfortunately we have to run the root node to know if you are calling the insights tool, or asking for something else.

Commit:
fixes

Commit:
Merge remote-tracking branch 'origin/max/onboarding' into max/onboarding

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

Commit:
Merge remote-tracking branch 'origin/master' into max/onboarding

Commit:
fix test

Commit:
Merge remote-tracking branch 'origin/max/onboarding' into max/onboarding

Commit:
fix: grammar error in prompt

Twixes commented on ee/hogai/graph/memory/nodes.py:
```diff
@@ -181,92 +224,138 @@ def should_process_message_chunk(cls, message: AIMessageChunk) -> bool:
     @property
     def _model(self):
-        return ChatOpenAI(model="gpt-4o-mini", temperature=0.3, disable_streaming=True, stop_sequences=["[Done]"])
+        return ChatOpenAI(model="o4-mini", temperature=0.3, disable_streaming=True, stop_sequences=["[Done]"])
```
Does this work for you? When I run onboarding, this fails for me, with the message that only `temperature=1.0` is supported by `o4-mini`, and `stop_sequences` aren't supported at all

kappa90 replied:
I switched back to 4.1, o3-mini wasn't working but I forgot to remove it from the commit

Commit:
fix: switch to gpt-4.1

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Twixes committed:
Merge branch 'master' into max/onboarding

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
Merge branch 'master' into max/onboarding

Commit:
fix: revert to gpt-4o for root / add langchain-perplexity

Commit:
fix: typescript error

Commit:
fix: bring back gpt-4o for memory collector

kappa90 commented:
@skoob13 I still need your approval on this otherwise it will never get merged 😅

Twixes commented:
You can also go full bias for action by dismissing the review! @kappa90 
<img width="840" alt="Screenshot 2025-05-23 at 15 01 16" src="https://github.com/user-attachments/assets/a9931a69-e895-4aa0-893b-0a142f97aff6" />

skoob13 reviewed (approved):
🚢

Commit:
Merge branch 'master' into max/onboarding
